TedTalkMVVM/TedTalkMVVM/view/DetailTedTalkViewController.swift

```diff
@@ -16,26 +16,56 @@ class DetailTedTalkViewController: UIViewController{
     @IBOutlet weak var descriptionDetailLabel: UILabel!
     @IBOutlet weak var tagDetailLabel: UILabel!
     @IBOutlet weak var webTedTalk: WKWebView!
+    @IBOutlet weak var previousTedTalkButton: UIButton!
+    @IBOutlet weak var nextTedTalkButton: UIButton!
     
-    var tedTalk: TedTalkDisplay? = nil
     
-    func setTedTalk(talk: TedTalkDisplay){
-        self.tedTalk = talk
+    @IBAction func previousDetailTedTalk(){
+        viewModel.getPreviousTedTalkDetail()
     }
+    
+    @IBAction func nextDetailTedTalk(){
+        viewModel.getNextTedTalkDetail()
+    }
+    
+    var viewModel: ViewModelProtocolDetail!
+    var index: Int = 0
+    
     override func viewDidLoad() {
         
         super.viewDidLoad()
-        
-        if let displayTedTalk = tedTalk{
-            
-            titleDetailLabel.text = displayTedTalk.title
-            webTedTalk.load(URLRequest(url: URL(string: displayTedTalk.url)!))
-            tagDetailLabel.text = "Tags: \(displayTedTalk.tags.joined(separator: ", "))"
-            let viewString = displayTedTalk.views as NSNumber
-            ofViewDetailLabel.text = "#of views: \(viewString.stringValue)"
-            nameDetailLabel.text = displayTedTalk.name
-            descriptionDetailLabel.text = displayTedTalk.description
-        }
+        viewModel = viewModelDetail(tedTalkManager: DisplayManagerTedTalk.sharedTedTalkManager,
+                                    observerTedTalk: { [unowned self] viewModel in
+                                        self.showTedTalkDetail(talk: viewModel.getTedTalkDetailsDisplay())
+                                    },observerNextTedTalkButton: { [unowned self] viewModel in
+                                        self.buttonNextDisable()}, observerPreviousTedTalkButton: { [unowned self] viewModel in
+                                            self.buttonPreviousDisable()}, observerTedTalkButtons: { [unowned self] viewModel in
+                                                self.buttonsEnables()})
```

Comment:
The closure for each button (next & previous) should return a boolean as a parameter, to indicate to the view if that button has to be enabled or not.
Also I do not think you need the third closure (observerTedTalkButtons), because take into consideration some border cases like index 0, which in that case only next button should be enabled, not previous and in the last ted talk of the array only previous button should be enabled not next one.
As it is now, you are enabling both buttons at the same time